pkg/controller/nodes/task/k8s/plugin_manager.go

```diff
@@ -116,12 +122,42 @@ func (e *PluginManager) LaunchResource(ctx context.Context, tCtx pluginsCore.Tas
 	AddObjectMetadata(tCtx.TaskExecutionMetadata(), o, config.GetK8sPluginConfig())
 	logger.Infof(ctx, "Creating Object: Type:[%v], Object:[%v/%v]", o.GroupVersionKind(), o.GetNamespace(), o.GetName())
 
-	err = e.kubeClient.GetClient().Create(ctx, o)
+	key := backoff.ComposeResourceKey(o)
+
+	pod, casted := o.(*v1.Pod)
+	if e.backOffController != nil && casted {
+		podRequestedResources := make(v1.ResourceList)
+
+		// Collect the resource requests from all the containers in the pod whose creation is to be attempted
+		// to decide whether we should try the pod creation during the back off period
+		for _, container := range pod.Spec.Containers {
```

Comment:
How about InitContainers? I've not thought through this.. It's more likely than not that containers will have higher limits than initContainers. So maybe it's ok?